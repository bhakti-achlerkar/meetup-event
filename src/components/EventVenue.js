import React from 'react';
import PropTypes from 'prop-types';

const EventVenue = ({ venue }) => {
  return (
    <div className="event-host">
      <span><i className="fas fa-map-marker fa-lg"></i></span>
      <div>
        <div> {venue.name}</div>
        <div className="event-address">
          <span>{venue.address_1}</span>
          <span>{ venue.address_2}</span>
        </div>
        <div className="event-address">{venue.city}, {venue.state} - {venue.zip}</div>
      </div>
    </div>
  );
};

EventVenue.propTypes = {
  venue: PropTypes.shape({
    name: PropTypes.string.isRequired,
    address_1: PropTypes.string,
    address_2: PropTypes.string,
    city: PropTypes.string,
    state: PropTypes.string,
    zip: PropTypes.string,
  })
};

export default EventVenue;
