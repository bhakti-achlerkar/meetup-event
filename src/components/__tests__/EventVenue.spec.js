import React from 'react';
import EventVenue from '../EventVenue';
import renderer from 'react-test-renderer';

const venue = {
  name: "VENUE_NAME",
  address_1: "Address1",
  address_2: "Address2",
  city: "CITY",
  state: "STATE",
  zip: "12344",
};

test('EventVenue rendered with event details', () => {
  const component = renderer.create(
    <EventVenue
      venue={venue}></EventVenue>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

