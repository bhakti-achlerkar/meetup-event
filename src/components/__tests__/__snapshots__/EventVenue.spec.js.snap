// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventVenue rendered with event details 1`] = `
<div
  className="event-host"
>
  <span>
    <i
      className="fas fa-map-marker fa-lg"
    />
  </span>
  <div>
    <div>
       
      VENUE_NAME
    </div>
    <div
      className="event-address"
    >
      <span>
        Address1
      </span>
      <span>
        Address2
      </span>
    </div>
    <div
      className="event-address"
    >
      CITY
      , 
      STATE
       - 
      12344
    </div>
  </div>
</div>
`;
